### PR TITLE
Adding more functionality for Linode cloud module

### PR DESCRIPTION
**Pull request by InTheCloudDan**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
cloud/linode

##### ANSIBLE VERSION
```
ansible 2.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

Added a number of features that allow for more customization when deploying a new Linode.
Options for:

- private IPs
- additional disks
- watchdog configurations

```

```

### CODE REVIEW

File: lib/ansible/modules/cloud/linode/linode.py
Comment (abadger) [inline lib/ansible/modules/cloud/linode/linode.py]: this is a version for the metadata format.  It should not be updated unless the format of the metadata changes.

**Review by abadger — DISMISSED** [review]

**Comment by InTheCloudDan** [comment]
changed the metadata version back to 1.0

**Comment by abadger** [comment]
@zbal, are you still around to review changes to the linode module?  I can review this PR for generic code quality but I do not use linode so I don't know if it makes sense from a high level.

File: lib/ansible/modules/cloud/linode/linode.py
Comment (abadger) [inline lib/ansible/modules/cloud/linode/linode.py]: stylistically, there should be some leading space here and probably should be broken onto a few lines.  Both of those will make this easier to read.

File: lib/ansible/modules/cloud/linode/linode.py
Comment (abadger) [inline lib/ansible/modules/cloud/linode/linode.py]: We'll need to update the version_added to either 2.3 or 2.4 depending on whether we can get this in before 2.3's module freeze date.  (A couple weeks from now).

**Comment by abadger** [comment]
I've taken a look for generic code quality and made some comments that should be easy to fix.  Beyond that, if zbal is still around I'd love to have his sign off on these being good changes since I can't test this or know if it makes sense myself.  Failing that, if you can find someone else who uses the linode module to review the changes as a user of those changes, I'd feel comfortable merging it.

We'll be freezing the modules for the 2.3 release in about 2 weeks, March 1st.  If we get everything in place by then this can be merged in time for 2.3.0 otherwise it can go in for 2.4.  (Note if everything gets done but I don't appear to notice, feel free to ping me on IRC, I'm abadger1999 on irc.freenode.net)

File: lib/ansible/modules/cloud/linode/linode.py
Comment (InTheCloudDan) [reply lib/ansible/modules/cloud/linode/linode.py]: I'll look into getting this updated.

**Comment by InTheCloudDan** [comment]
@abadger If it helps any, I work for Linode and have been using this internally without an issue. Not saying there may not be any issues but I haven't run into them so far.

**Comment by abadger** [comment]
@InTheCloudDan Cool.  I'd be willing to merge this once you update then.  Also, if you'd like to become a comaintainer for the linode module, you can open a pull request on the ansibulbot repository https://github.com/ansible/ansibullbot  ( Look at some of hte existing maintainer PRs for what to modify: https://github.com/ansible/ansibullbot/pulls )

**Comment by abadger** [comment]
Rebased, squashed, and merged here: https://github.com/ansible/ansible/commit/320c791cfbb56e9b118b00baddfbb71a1e6fc7cb

Thanks!

**Pull request closed without merging**
